### PR TITLE
EPMDJ-4840: Add tooltip into overflow-text component

### PR DESCRIPTION
**Pull request by IliaSolovev**
Issue: EPMDJ-4840

### CODE REVIEW

File: src/components/overflow-text/overflow-text.tsx [hunk 1]
```diff
@@ -1,5 +1,40 @@
...
+export interface Props {
+  className?: string;
+  children: React.ReactNode | React.ReactNode[];
+  tooltipText?: string;
+}
+
+const overflowText = BEM(styles);
+
+export const OverflowText = overflowText(({ className, children, tooltipText }: Props) => {
+  const rootRef = React.useRef<HTMLDivElement>(null);
```
Comment (RomanAchilov) [inline src/components/overflow-text/overflow-text.tsx:15]: childrenRef

File: src/components/overflow-text/overflow-text.tsx [hunk 2]
```diff
@@ -1,5 +1,40 @@
...
+  const childrenNode = React.useRef<HTMLDivElement>(null);
+  const [isIntersecting, setIsIntersecting] = React.useState(true);
```
Comment (RomanAchilov) [inline src/components/overflow-text/overflow-text.tsx:17]: const [hint, setHint] = React.useState('');

File: src/components/overflow-text/overflow-text.tsx [hunk 3]
```diff
@@ -1,5 +1,40 @@
...
+
+  React.useEffect(() => {
+    const observer = new IntersectionObserver(
+      ([entry]) => setIsIntersecting(entry.isIntersecting),
```
Comment (RomanAchilov) [inline src/components/overflow-text/overflow-text.tsx:21]: ([entry]) => !entry.isIntersecting && setHint(children),

File: src/components/overflow-text/overflow-text.tsx [hunk 4]
```diff
@@ -1,5 +1,40 @@
-import { BEM } from '@redneckz/react-bem-helper';
+import { BEM, span } from '@redneckz/react-bem-helper';
 
+import * as React from 'react';
 import styles from './overflow-text.module.scss';
 
-export const OverflowText = BEM(styles)('div');
+export interface Props {
+  className?: string;
+  children: React.ReactNode | React.ReactNode[];
```
Comment (RomanAchilov) [inline src/components/overflow-text/overflow-text.tsx:8]: string

File: src/components/overflow-text/overflow-text.tsx [hunk 5]
```diff
@@ -1,5 +1,40 @@
...
+  tooltipText?: string;
```
Comment (RomanAchilov) [inline src/components/overflow-text/overflow-text.tsx:9]: delete

File: src/components/overflow-text/overflow-text.tsx [hunk 6]
```diff
@@ -1,5 +1,40 @@
...
+  React.useEffect(() => {
+    const observer = new IntersectionObserver(
+      ([entry]) => setIsIntersecting(entry.isIntersecting),
+      {
+        root: rootRef.current,
+        threshold: 1.0,
+      },
+    );
+
+    childrenNode.current !== null && observer.observe(childrenNode.current);
```
Comment (RomanAchilov) [inline src/components/overflow-text/overflow-text.tsx:28]: childrenRef.current && observer.observe(childrenRef.current);

File: src/components/overflow-text/overflow-text.tsx [hunk 7]
```diff
@@ -1,5 +1,40 @@
...
+
+    return () => {
+      observer.disconnect();
+    };
+  }, []);
+
+  return (
+    <div className={className} ref={rootRef}>
```
Comment (RomanAchilov) [inline src/components/overflow-text/overflow-text.tsx:36]: delete ref={rootRef}

File: src/components/overflow-text/overflow-text.tsx [hunk 8]
```diff
@@ -1,5 +1,40 @@
...
+      <span ref={childrenNode} title={!isIntersecting ? tooltipText : ''}>{children}</span>
```
Comment (RomanAchilov) [inline src/components/overflow-text/overflow-text.tsx:37]: <span ref={messageRef} title={hint}>{children}</span>

File: src/components/overflow-text/overflow-text.stories.tsx [hunk 9]
```diff
@@ -5,6 +5,6 @@ import { OverflowText } from './overflow-text';
 
 storiesOf('Overflowext', module).add('Overflow text', () => (
   <div style={{ width: '150px' }}>
-    <OverflowText>Overflow text Overflow text</OverflowText>
+    <OverflowText tooltipText="Overflow text">Overflow text Overflow text</OverflowText>
```
Comment (RomanAchilov) [inline src/components/overflow-text/overflow-text.stories.tsx:8]: more examples Pls

**Review by RomanAchilov — CHANGES_REQUESTED** [review]

File: src/components/overflow-text/overflow-text.tsx [hunk 10]
```diff
@@ -1,5 +1,38 @@
-import { BEM } from '@redneckz/react-bem-helper';
+import { BEM, span } from '@redneckz/react-bem-helper';
```
Comment (RomanAchilov) [inline src/components/overflow-text/overflow-text.tsx:1]: unused span

**Review by RomanAchilov — CHANGES_REQUESTED** [review]
use data-title for styling

File: src/components/overflow-text/overflow-text.tsx [hunk 11]
```diff
@@ -1,5 +1,42 @@
 import { BEM } from '@redneckz/react-bem-helper';
 
+import * as React from 'react';
```
Comment (RomanAchilov) [inline src/components/overflow-text/overflow-text.tsx:3]: nice

File: src/components/overflow-text/overflow-text.tsx [hunk 12]
```diff
@@ -1,5 +1,42 @@
...
+
+    messageRef.current && observer.observe(messageRef.current);
+
+    return () => {
+      observer.disconnect();
+    };
+  }, []);
+
+  return (
+    <div className={className} data-title={hint || null}>
```
Comment (RomanAchilov) [inline src/components/overflow-text/overflow-text.tsx:34]: null?

**Review by RomanAchilov — CHANGES_REQUESTED** [review]

File: src/components/overflow-text/overflow-text.tsx (hunk 12 above)
Comment (IliaSolovev) [reply src/components/overflow-text/overflow-text.tsx]: It need that when we pass there empty string we don`t see empty tooltip block with paddings

**Pull request closed without merging**